src/test/java/com/yahoo/bullet/bql/parser/BQLParserTest.java

```diff
@@ -10,370 +10,21 @@
-    public void testExpressionFormatterVisitNode() {
-        new ExpressionFormatter.Formatter(Optional.empty()).process(simpleWithQuery());
-    }
-
-    @Test
-    public void testExpressionFormatterConstructor() {
-        assertTrue(new ExpressionFormatter() instanceof ExpressionFormatter);
-    }
-
-    @Test(expectedExceptions = IllegalArgumentException.class, expectedExceptionsMessageRegExp = "\\QExpression inside formatExpression() must not be null\\E.*")
-    public void testNullExpression() {
-        formatExpression(null, Optional.empty());
-    }
-
-    @Test(expectedExceptions = ParsingException.class, expectedExceptionsMessageRegExp = "\\Qline 1:1: Unexpected decimal literal\\E.*")
-    public void testRejectParsingOptions() {
-        BQL_PARSER.createExpression("10.5", new ParsingOptions(REJECT));
-    }
-
-    @Test(expectedExceptions = UnsupportedOperationException.class, expectedExceptionsMessageRegExp = "\\QNot yet implemented\\E.*")
-    public void testAggregateResultNullNext() {
-        new ASTBuilder(new ParsingOptions()).aggregateResult(identifier("aaa"), null);
-    }
-
-    @Test(expectedExceptions = UnsupportedOperationException.class, expectedExceptionsMessageRegExp = "\\QNot yet implemented\\E.*")
-    public void testAggregateResultNullAggregate() {
-        new ASTBuilder(new ParsingOptions()).aggregateResult(identifier("aaa"), identifier("bbb"));
-    }
-
-    @Test(expectedExceptions = ParsingException.class, expectedExceptionsMessageRegExp = "\\Qline 1:3: exception as expected\\E.*")
-    public void testParsingException() {
-        NodeLocation location = new NodeLocation(1, 1);
-        throw new ParsingException("exception as expected", location);
-    }
-
-    private static void assertStatement(String query, Statement expected) {
-        assertParsed(query, expected, BQL_PARSER.createStatement(query, new ParsingOptions()));
-        assertFormattedBQL(BQL_PARSER, expected);
-    }
-
-    private static void assertExpression(String expression, Expression expected) {
-        assertParsed(expression, expected, BQL_PARSER.createExpression(expression, new ParsingOptions()));
-    }
-
-    private static void assertParsed(String input, Node expected, Node parsed) {
-        if (!parsed.equals(expected)) {
-            fail(format("expected\n\n%s\n\nto parse as\n\n%s\n\nbut was\n\n%s\n",
-                    indent(input),
-                    indent(formatBQL(expected, Optional.empty())),
-                    indent(formatBQL(parsed, Optional.empty()))));
-        }
-    }
-
-    private static String indent(String value) {
-        String indent = "    ";
-        return indent + value.trim().replaceAll("\n", "\n" + indent);
+        QueryNode node = parser.createQueryNode("SELECT all FROM STREAM()");
+        Assert.assertEquals(node.getSelect().getSelectItems().get(0).getExpression().getName(), "all");
```

Comment:
You can probably keep some of the tests in here

Reply:
Maybe. I would have to add in the methods to create parts of the query node (i.e. parse just an expression), which would be useful for just unit testing I guess.